frontend/public/components/storage/create-pvc.tsx

```diff
@@ -0,0 +1,368 @@
+            selectedKey={storageClass}
+            namespace={namespace}
+            required={false}
+            name="storageClass"
+            placeholder="Select storage class"
+          />
+          {storageClass && <Clear />}
+        </div>
+        <p className="form-text text-muted" id="storageclass-dropdown-help">
+          Optional: Select storage class to associate with this new claim.
+        </p>
+        <label className="control-label co-required" htmlFor="pvc-name">
+          Name
+        </label>
+        <div className="form-group">
+          <input
+            className="form-control"
+            type="text"
+            onChange={this.handleChange}
+            placeholder="my-storage-claim"
+            aria-describedby="volume-name-help"
+            id="pvc-name"
+            name="pvcName"
+            pattern="[a-z0-9](?:[-a-z0-9]*[a-z0-9])?"
+            required
+          />
+        </div>
+        <p className="form-text text-muted" id="pvc-name-help">
+          A unique name for the storage claim within the project.
+        </p>
+        <label className="control-label co-required" htmlFor="access-mode">
+          Access Mode
+        </label>
+        <div className="form-group">
+          {this.state.radios.map(radio => {
+            const checked = radio.value === this.state.accessMode;
+            return (
+              <RadioInput
+                {...radio}
+                key={radio.value}
+                onChange={this.handleChange}
+                checked={checked}
+                aria-describedby="access-mode-help"
+                className="radio-inline"
+                name="accessMode"
+              />
+            );
+          })}
+          <p className="form-text text-muted" id="access-mode-help">
+            Permissions to the mounted drive.
+          </p>
+        </div>
+        <label className="control-label co-required" htmlFor="request-size-input">
+          Size
+        </label>
+        <RequestSizeInput
+          name="requestSize"
+          required={false}
+          onChange={this.handleChange}
+          placeholder={requestSizeValue}
+          defaultRequestSizeUnit={requestSizeUnit}
+          dropdownUnits={dropdownUnits}
+          ariaName="requestSize-helpID"
+        />
+        <p className="form-text text-muted" id="requestSize-help">
+          Desired storage capacity.
+        </p>
+        <Checkbox
+          label="Use label selectors to request storage"
+          onChange={this.toggleShowLabelSelectors}
+          checked={showLabelSelectors}
+          name="showLabelSelector"
+        />
+        <div className="form-group">
+          {showLabelSelectors && (
+            <NameValueEditor
+              nameValuePairs={nameValuePairs}
+              valueString="Selector"
+              nameString="Label"
+              addString="Add Value"
+              readOnly={false}
+              allowSorting={false}
```

Comment:
I believe the label selectors get sorted by the API save and doesn't match how we show them on the UI.  Since we cannot control the ordering, I felt there was no advantage to allow sorting.

Reply:
Right, order isn't preserved by the API here, so there's no real reason to allow ordering.